_posts/2017-03-07-hddtools.md

```diff
@@ -0,0 +1,323 @@
+
+
+```r
+# Filter GRDC catalogue based on a country code
+GRDC_IT <- catalogueGRDC(columnName = "country_code", columnValue = "IT")
+
+# Convert the table to a SpatialPointsDataFrame
+hydro <- SpatialPointsDataFrame(coords = GRDC_IT[, c("long", "lat")], 
+                                data = GRDC_IT)
+
+# Plot the stations on the map
+plot(Italy, col = NA, border = "darkgrey")
+plot(hydro, add = TRUE, col = "blue", pch = 1)
+```
+
+![](/assets/blog-images/2013-03-07-hddtools/unnamed-chunk-7-1.png)
+
+The arguments `columnName` and `columnValue` can be used to filter over other columns. For instance, the example below shows how to subset stations along the Tanaro River which source in the Ligurian Alps.
+
+
+```r
+# Filter GRDC catalogue based on rivername, search is not case sensitive!
+Tanaro <- catalogueGRDC(columnName = "river", columnValue = "Tanaro")
+
+# Convert the table to a SpatialPointsDataFrame
+TanaroSP <- SpatialPointsDataFrame(coords = Tanaro[, c("long", "lat")], 
+                                   data = Tanaro)
+
+# Highight in red the stations on the Tanaro River
+plot(Italy, col = NA, border = "darkgrey")
+plot(hydro, add = TRUE, col = "blue", pch = 1)
+plot(TanaroSP, add = TRUE, col = "red", pch = 1)
+```
+
+![](/assets/blog-images/2013-03-07-hddtools/unnamed-chunk-8-1.png)
+
+If `columnName` refers to a numeric field, in the `columnValue` other terms of comparison can be specified. For example, it is straightforward to find out that the country with the most longstanding monitoring stations is Germany (with 11 out of 15 stations with more than 150 years of recordings).
+
+
+```r
+# Filter GRDC stations with more than 150 years of recordings
+GRDC150 <- catalogueGRDC(columnName = "d_yrs", columnValue = ">150")
+
+# Which country has the most longstanding monitoring stations?
+table(GRDC150$country_code)
+```
+
+```
+## 
+## DE FI LT RO SE 
+## 11  1  1  1  1
+```
+
+```r
+# Where is the oldest station?
+oldest <- GRDC150[which(GRDC150$d_yrs == max(GRDC150$d_yrs)), 
+                  c("grdc_no", "river", "station", "d_yrs")]
+oldest
+```
+
+```
+## # A tibble: 1 × 4
+##   grdc_no      river station d_yrs
+##     <chr>      <chr>   <chr> <dbl>
+## 1 6340120 ELBE RIVER DRESDEN   208
+```
+
+The oldest stastion is in Dresden on the Elbe River. Let's now find out whether monthly data is available. For this we need the GRDC identification number (this is stored in the column grdc_no of the catalogue) and the function `tsGRDC()`.  
+
+
+```r
+# Monthly data extraction
+DresdenStation <- tsGRDC(stationID = oldest$grdc_no, plotOption = TRUE)
+```
+
+![](/assets/blog-images/2013-03-07-hddtools/unnamed-chunk-10-1.png)<!-- -->
```

Comment:
In this plot and some of the others I had to think hard about the meaning of the axis labels. Rather than having to modify the plots, could you add something in the text? 

- For the Dresden plot, that m3/s indicates flow? 
- For TRMM.1 TRMM.2 TRMM.3, perhaps explain what the fig(s) show and what are each of those three
- For the Morwick and Broad River plots, same idea e.g. what are Q, P, E etc